### PR TITLE
fix(schema): plugin level schema validation

### PR DESCRIPTION
**Pull request by willsoto**
## Description

This PR fixes plugin level schema validation. Changes themselves were minimal, but the `config.schema.json` itself seems to have a lot of unrelated changes (mostly formatting) not sure why, maybe it hasn't been regenerated in a while.

Related # (issue)

#6314

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots/Sandbox (if appropriate/relevant):

https://user-images.githubusercontent.com/2791730/132767685-ea67f300-d875-4985-be57-f621c9be0f87.mp4

## How Has This Been Tested?

Manually tested using `alt-codegen.yml` - couldn't find any automated tests but willing to add those if they are present.

**Test Environment**:
- OS: MacOS
- `@graphql-codegen/...`:  `master`
- NodeJS: 16.8.0

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

Opening this to get a conversation started. Happy to change implementation if people have better ideas.

### CODE REVIEW

File: packages/utils/config-schema/src/index.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { generateDocs } from './docs';
 import mkdirp from 'mkdirp';
 import { pluginsConfigurations, presetsConfigurations } from './plugins';
 import { join } from 'path';
-import { apply } from 'jsonpath';
+import jsonpath from 'jsonpath';
```
Comment (willsoto) [inline packages/utils/config-schema/src/index.ts:8]: Changed due to [this issue](https://github.com/dchester/jsonpath/issues/126)

**Comment by dotansimha** [comment]
@B2o5T some of the changes are relevant to the latest changes I pushed to the new website branch. Can you please try to cherry-pick / apply these there (I guess in `transform.ts`)

**Comment by dotansimha** [comment]
Thanks @willsoto , I'm closing this PR, since @B2o5T cherry-picked your changes into the new website setup we are building.

**Comment by B2o5T** [comment]
@willsoto I made cherry-pick of your commit and resolved conflicts, could you test if everything working in this PR https://github.com/dotansimha/graphql-code-generator/pull/7010 ? And thanks so much!

**Pull request closed without merging**